packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -100,37 +101,53 @@ class _SnapshotListTitle extends StatelessWidget {
   }
 }
 
-class _SnapshotListItems extends StatelessWidget {
-  _SnapshotListItems({Key? key, required this.controller}) : super(key: key);
+class _SnapshotListItems extends StatefulWidget {
+  const _SnapshotListItems({Key? key, required this.controller})
+      : super(key: key);
 
   final DiffPaneController controller;
-  final headerHeight = 1.20 * defaultRowHeight;
+
+  @override
+  State<_SnapshotListItems> createState() => _SnapshotListItemsState();
+}
+
+class _SnapshotListItemsState extends State<_SnapshotListItems>
+    with AutoDisposeMixin {
+  final _headerHeight = 1.20 * defaultRowHeight;
+  final _scrollController = ScrollController();
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    addAutoDisposeListener(widget.controller.snapshots);
+    addAutoDisposeListener(widget.controller.selectedIndex);
+    addAutoDisposeListener(widget.controller.scrollToLastSnapshot, () {
```

Comment:
we don't need to create a scrollToLastSnapshot notifier to do this. We can do this using the snapshots listener.
```
late int currentSnapshotsLength;

void didChangeDependencies() {
  currentSnapshotsLength = widget.controller.snapshots.length;
  addAutoDisposeListener(widget.controller.snapshots, {
    final newSnapshotsLength = widget.controller.snapshots.length;
    if (newSnapshotsLength > currentSnapshotsLength) {
      setState(() {
        _scrollController.autoScrollToBottom();
      }
    }
    currentSnapshotsLength =  newSnapshotsLength;
  });
} 
```